### PR TITLE
Adds an editable build of arches templating

### PR DESCRIPTION
**Pull request by aarongundel**
Adds an editable build of arches templating

### CODE REVIEW

**Review by jacobtylerwalls — APPROVED** [review]
Luckily, `arches-templating` doesn't define any arches-related requirements, so we won't end up with the same headache of needing to uninstall arches afterward.

File: Dockerfile [hunk 1]
```diff
@@ -68,13 +68,16 @@ RUN rm -rf /root/.cache/pip/*
 # FIXME: ADD from github repository instead?
 COPY ./arches ${ARCHES_ROOT}
 COPY ./arches-for-science ${AFS_ROOT}
-
+COPY ./arches_templating ${TEMPLATING_ROOT}
```
[the remark this answers is not in the document]
Comment (aarongundel) [reply Dockerfile]: @jacobtylerwalls this is a good point.  I've fixed that, I think the default directory name should be whatever the repo is. 
 For me, when I originally wrote templating, it was arches_templating all the way through.  That changed when we put it on pypi, I just hadn't updated it locally/anywhere else.  For the sake of completeness, I also decided to try some experiments with build args to give the dev a little more control over their setup.  I know we talked about that a little in another PR, and my comments there still apply, but I noticed that the base paths were hardcoded in the Dockerfile, but not in the compose file.  That makes the compose variables useful for _running_ the container, but _useless_ for building it.  And unfortunately the way this is handled in compose is more than a bit obtuse, requiring a couple different methods to set the variables in question - I've noted how they can be set in the compose file.  If you want to try it, cool - if not, it should just work out of the box and give everyone a bit more flexibility if needed in the future.

[the remark this answers is not in the document]
Comment (aarongundel) [reply Dockerfile]: I'm going to push this back to you - I want to make sure it at least works on someone else's machine before I merge it.

**Review by jacobtylerwalls — APPROVED** [review]
Tested, works well!

Tested the new variables, too. Didn't wait for the whole build to succeed, but I got past the failure I would have seen if I'd still had `arches_templating/` instead of `arches-templating`, like so:

```sh
docker compose build --build-arg="ARCHES_TEMPLATING_HOST_DIR=../arches_templating"
```

Nice work!

**Pull request merged**